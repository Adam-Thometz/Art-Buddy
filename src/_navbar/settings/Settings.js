import React from 'react';
import useRoster from '../../_hooks/useRoster';

import { useSelector, useDispatch } from 'react-redux';
import { setRoster, setVolume, toggleColorBlind, toggleTextToSpeech } from '../../_redux/actions/mainSettingsActions';

import './Settings.css';

import Icon from '../../_components/icon/Icon';
import Toggle from '../../_components/toggle/Toggle';
import Dropdown from '../../_components/dropdown/Dropdown';
import Popup from '../../_components/popup/Popup';
import NewRoster from './rosters/NewRoster';

import volumeHigh from './_icons/volume/volume-high.png';
import volumeMid from './_icons/volume/volume-medium.png';
import volumeLow from './_icons/volume/volume-low.png';

const Settings = () => {
  const {
    volume,
    textToSpeech,
    colorBlind,
    roster
  } = useSelector(state => state.mainSettings);
  const [rosters] = useRoster();
  const dispatch = useDispatch();
  
  const createRosterSelection = () => {
    const rosterSelection = {};
    for (let rosterId in rosters) {
      rosterSelection[rosterId] = rosters[rosterId].name;
    };
    return rosterSelection;
  };
  
  const handleVolume = e => dispatch(setVolume(+e.currentTarget.id));
  const handleTextToSpeech = () => dispatch(toggleTextToSpeech());
  const handleColorBlind = () => dispatch(toggleColorBlind());
  const handleSetRoster = e => dispatch(setRoster(rosters[e.target.id]));

  const selectedVolume = { 
    filter: 'invert(55%) sepia(55%) saturate(5241%) hue-rotate(163deg) brightness(97%) contrast(102%)'
  };

  return (
    <main className='Settings'>
      <span className='Settings-text'>Volume</span>
      <div className='Settings-icons-volume'>
        <Icon
          icon={volumeLow}
          otherImgStyles={volume === -12 ? selectedVolume : {}}
          text='LOW'
          onClick={handleVolume}
          id='-12'
          size='50px' width='40px'
        />
        <Icon
          icon={volumeMid}
          otherImgStyles={volume === 0 ? selectedVolume : {}}
          text='MEDIUM'
          onClick={handleVolume}
          id='0'
          size='50px'
        />
        <Icon
          icon={volumeHigh}
          otherImgStyles={volume === 12 ? selectedVolume : {}}
          text='HIGH'
          onClick={handleVolume}
          id='12'
          size='50px' width='60px'
        />
      </div>

      <span className='Settings-text'>Text-to-Speech</span>
      <span className='Settings-toggle'>
        <Toggle currToggle={textToSpeech} toggleFn={handleTextToSpeech} />
      </span>

      <span className='Settings-text'>Color Blind Mode</span>
      <span className='Settings-toggle'>
        <Toggle currToggle={colorBlind} toggleFn={handleColorBlind} />
      </span>

      <span className='Settings-text'>Roster</span>
      <div className='Settings-icons-roster'>
        <Popup
          title='NEW ROSTER'
          trigger='+'
          triggerClass='Settings-add-roster'
          popup={<NewRoster />}
        />
        {Object.keys(rosters).length ? (
          <Dropdown labelText={roster.name ? roster.name : 'SELECT ROSTER'} onClick={handleSetRoster} options={createRosterSelection()} />
        ) : null}
      </div>
    </main>
  );
};

export default Settings;