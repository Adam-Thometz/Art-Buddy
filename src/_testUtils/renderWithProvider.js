import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;