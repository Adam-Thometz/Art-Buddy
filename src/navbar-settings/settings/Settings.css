.Settings {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(4, 1fr);
  width: 70%;
  margin: 0 auto;
  height: 60%;
  font-size: 2.5rem;
}

.Settings-text {
  text-align: right;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  padding-right: 50px;
}