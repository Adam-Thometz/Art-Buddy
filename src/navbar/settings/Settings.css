.Settings {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(4, 1fr);
  width: 90%;
  margin: 0 auto;
  justify-content: space-evenly;
  height: 60%;
  font-size: 2.5rem;
}

.Settings-text {
  text-align: right;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  padding-right: 50px;
}

.Settings-icons {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: flex-start;
}

.Settings-icons-volume {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 25vw;
}

.Settings-icons-volume img {
  cursor: pointer;
}

.Settings-toggle {
  display: flex;
  align-items: center;
}

.Settings-icons-roster {
  display: flex;
  align-items: center;
}

.Settings-add-roster {
  font-size: 80px;
  margin: 20px;
  cursor: pointer;
}