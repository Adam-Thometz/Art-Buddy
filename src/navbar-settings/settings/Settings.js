import "./Settings.css";

import Volume from "./volume/Volume";
import TextToSpeech from "./text-to-speech/TextToSpeech";
import ColorBlind from "./color-blind/ColorBlind";
import Rosters from "./rosters/Rosters";

const Settings = () => {
  return (
    <main className="Settings">
      <p className="Settings-text">Volume</p>
      <Volume />

      <p className="Settings-text">Text-to-Speech</p>
      <TextToSpeech />

      <p className="Settings-text">Color Blind Mode</p>
      <ColorBlind />

      <p className="Settings-text">Roster</p>
      <Rosters />
    </main>
  );
};

export default Settings;
