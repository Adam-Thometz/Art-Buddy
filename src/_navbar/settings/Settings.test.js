import React from "react";
import renderWithProvider from "../../_testUtils/renderWithProvider";
import Settings from './Settings';

describe('Settings component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<Settings />);
  });

  it('matches the snapshot of the Settings', () => {
    const { asFragment } = renderWithProvider(<Settings />);
    expect(asFragment()).toMatchSnapshot();
  });
});