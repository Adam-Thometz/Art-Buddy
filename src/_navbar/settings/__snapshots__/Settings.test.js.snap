// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component matches the snapshot of Settings 1`] = `
<DocumentFragment>
  <main
    class="Settings"
  >
    <span
      class="Settings-text"
    >
      Volume
    </span>
    <div
      class="Settings-icons-volume"
    >
      <figure
        class="Icon"
        id="-12"
      >
        <img
          alt=""
          src="file"
          style="height: 50px; width: 40px; cursor: pointer;"
        />
        <figcaption
          class="Icon-text"
          style="font-size: 1rem;"
        >
          LOW
        </figcaption>
      </figure>
      <figure
        class="Icon"
        id="0"
      >
        <img
          alt=""
          src="file"
          style="height: 50px; width: 50px; cursor: pointer; filter: invert(55%) sepia(55%) saturate(5241%) hue-rotate(163deg) brightness(97%) contrast(102%);"
        />
        <figcaption
          class="Icon-text"
          style="font-size: 1rem;"
        >
          MEDIUM
        </figcaption>
      </figure>
      <figure
        class="Icon"
        id="12"
      >
        <img
          alt=""
          src="file"
          style="height: 50px; width: 60px; cursor: pointer;"
        />
        <figcaption
          class="Icon-text"
          style="font-size: 1rem;"
        >
          HIGH
        </figcaption>
      </figure>
    </div>
    <span
      class="Settings-text"
    >
      Text-to-Speech
    </span>
    <span
      class="Settings-toggle"
    >
      <div
        class="Toggle"
      >
        <div
          class="Toggle-switch off"
        >
          <div
            class="Toggle-button"
          />
          <span>
            ON
          </span>
          <span>
            OFF
          </span>
        </div>
      </div>
    </span>
    <span
      class="Settings-text"
    >
      Color Blind Mode
    </span>
    <span
      class="Settings-toggle"
    >
      <div
        class="Toggle"
      >
        <div
          class="Toggle-switch off"
        >
          <div
            class="Toggle-button"
          />
          <span>
            ON
          </span>
          <span>
            OFF
          </span>
        </div>
      </div>
    </span>
    <span
      class="Settings-text"
    >
      Roster
    </span>
    <div
      class="Settings-icons-roster"
    >
      <div
        class="Settings-add-roster"
      >
        <figure
          class="Icon"
        >
          <img
            alt=""
            src="file"
            style="height: 42px; width: 42px;"
          />
          <figcaption
            class="Icon-text"
            style="font-size: 1rem;"
          >
            ADD
          </figcaption>
        </figure>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
